janusgraph-test/src/main/java/org/janusgraph/graphdb/JanusGraphIoTest.java

```diff
@@ -60,5 +111,62 @@ public void testGeoShapeSerializationReadWriteAsGryo() throws Exception {
         graph.io(IoCore.gryo()).readGraph("/tmp/test.kryo");
 
         JanusGraphIndexTest.assertGraphOfTheGods(graph);
+        if (makeGeoshape != null) {
+            assertGeoshape(makeGeoshape);
+        }
+    }
+
+    private void addGeoshape(Function<Geoshape,Geoshape> makeGeoshape) {
+        JanusGraphTransaction tx = graph.newTransaction();
+        graph.traversal().E().has("place").toList().stream().forEach(e-> {
+            Geoshape place = (Geoshape) e.property("place").value();
+            e.property("shape", makeGeoshape.apply(place));
+        });
+        tx.commit();
+    }
+
+    private void assertGeoshape(Function<Geoshape,Geoshape> makeGeoshape) {
+        graph.traversal().E().has("place").toList().stream().forEach(e-> {
+            assertTrue(e.property("shape").isPresent());
+            Geoshape place = (Geoshape) e.property("place").value();
+            Geoshape expected = makeGeoshape.apply(place);
+            Geoshape actual = (Geoshape) e.property("shape").value();
+            assertEquals(expected, actual);
+        });
     }
+
+    private Function<Geoshape,Geoshape> makePoly = place -> {
```

Comment:
static final? it seems like these functions dont depend on member state, so lets make it clear